src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.barclay.argparser;
+
+/**
+ * Exceptions thrown by CommandLineParser implementations.
+ */
+public class CommandLineException extends RuntimeException {
+    private static final long serialVersionUID = 1L;
+
+    public CommandLineException( String msg ) {
+        super(msg);
+    }
+
+    public CommandLineException( String message, Throwable throwable ) {
+        super(message, throwable);
+    }
+
+    // todo -- fix up exception cause passing
+    public static class MissingArgument extends CommandLineException {
+        private static final long serialVersionUID = 0L;
+
+        public MissingArgument(String arg, String message) {
+            super(String.format("Argument %s was missing: %s", arg, message));
+        }
+    }
+
+    //TODO: should this be a CommandLineException ? It was previously
```

Comment:
I think it makes sense for this to be a commandline exception even if it's usually thrown by a tool during it's initialization.